zerver/tests/test_event_system.py

```diff
@@ -1157,7 +1157,7 @@ def test_queries(self) -> None:
 
         self.login_user(user)
 
-        with self.assert_database_query_count(42):
+        with self.assert_database_query_count(43):
```

Comment:
This query increase is due to calling `is_user_in_group` for computing `can_create_public_streams`.

Reply:
Makes sense. We should probably have a #backend thread about how to optimize this, since adding 1 query per setting here will eventually be a problem.

There's basically two theories:
- We can add a client capability for it not needing the `can_create_public_streams` values; this would mean we only spend these queries for simpler clients.
- We can compute these by walking the set of user groups that we've already fetched for the user for that part of the API, in the common case that we also fetched user group data.

Reply:
We pass these `can_....` fields only for a few settings, so might be fine as there would not be an extra query for each group setting.

But we can probably add a client capability. I am not sure but we probably discussed about removing these fields, but yeah it was not recent and has been some time.

Reply:
Yeah. Can you open an issue to track it and put it on the groups board, since it might not be an immediate follow-up?

Reply:
Opened #30383. But am not sure if we should include it in the groups board, since it is not directly related to the groups permissions framework.
But yeah, we can add it since the extra queries are added as a result of migrating them to groups framework.